src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -284,52 +300,88 @@ protected override void StopProcessing()
             {
                 return;
             }
 
-            TcpClient client = new TcpClient();
+            int successfulConnections = 0;
 
-            try
+            for (var i = 1; i <= Count; i++)
             {
-                Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
-                string targetString = targetAddress.ToString();
-
-                for (var i = 1; i <= TimeoutSeconds; i++)
+                TcpTestStatus testResult = new TcpTestStatus(
+                    i,
+                    Source,
+                    targetNameOrAddress,
+                    targetAddress.ToString(),
+                    TcpPort,
+                    0,
+                    TcpTestStatus.TcpConnectionTestResult.New
+                );
+                
+                try
                 {
-                    Task timeoutTask = Task.Delay(millisecondsDelay: 1000);
+                    TcpClient client = new TcpClient();
+                    
+                    Stopwatch stopwatch = new Stopwatch();
+                    stopwatch.Start();
+
+                    Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
+                    Task timeoutTask = Task.Delay(new TimeSpan(0, 0, TimeoutSeconds));
+
                     Task.WhenAny(connectionTask, timeoutTask).Result.Wait();
+                    stopwatch.Stop();
+
+                    client.Close();
 
                     if (timeoutTask.Status == TaskStatus.Faulted || timeoutTask.Status == TaskStatus.Canceled)
                     {
-                        // Waiting is interrupted by Ctrl-C.
-                        WriteObject(false);
+                        testResult.Result = TcpTestStatus.TcpConnectionTestResult.Cancelled;                        
                         return;
                     }
-
+                    
+                    if (timeoutTask.Status == TaskStatus.RanToCompletion)
+                    {
+                        testResult.Result = TcpTestStatus.TcpConnectionTestResult.Timeout;
+                    }
+                    
                     if (connectionTask.Status == TaskStatus.RanToCompletion)
+                    {
+                        successfulConnections++;
+                        testResult.Result = TcpTestStatus.TcpConnectionTestResult.Success;
+                        testResult.Latency = stopwatch.ElapsedMilliseconds;
+                    }
+                }
+                catch
+                {
+                    testResult.Result = TcpTestStatus.TcpConnectionTestResult.Failed;
+                }                  
+
+                if (Quiet.IsPresent)
+                {   
+                    if (i > successfulConnections)
+                    {
+                        WriteObject(false);
+                        return;
+                    }
+                    else if(i == Count)
```

Comment:
This else isn't needed, since you're returning in the if branch. You can just output and return here without an additional check.

Reply:
The reason I added this logic was to test if we were on the last loop. When using the `-quiet` option with `-count`, it will only return true if all tests completed successfully. Otherwise, it'll return false upon the first failure. 

Do you reckon this is a good flow, or would it be better to return the result of the first test when using `-quiet`?

Reply:
We should follow ping mode but we have a bug there.
From docs https://docs.microsoft.com/en-us/powershell/module/Microsoft.PowerShell.Management/Test-Connection?view=powershell-6
> If any ping succeeds, $True is returned.
If all pings fail, $False is returned.

Reply:
Looking at the code, I think that was carried over from the original Test-Connection implementation. We should fix that too, but probably in a separate PR. 

Unless it makes sense to align both of them in the same PR.

If we follow the documentation, we should just return `true` on any successful tests, and only return `false` if all pings / TCP connections fail.

Reply:
I see what you guys mean, I can definitely change the logic to match. I'll jump on that as soon as I can!

Reply:
Updated the logic! 

When using `-Quiet`, if any connection is successful, we return `true`. Otherwise, if we get to the end of the count with all failures, we return `false`